### PR TITLE
Add missing CanReAttest agent listing filter and output mask

### PR DESCRIPTION
**Pull request by guilhermocc**
Ongoing work for https://github.com/spiffe/spire/issues/1836

### CODE REVIEW

**Comment by faisal-memon** [comment]
i can review

File: proto/spire/api/server/agent/v1/agent.proto [hunk 1]
```diff
@@ -77,6 +77,9 @@ message ListAgentsRequest {
 
         // Filters agents to those that are banned.
         google.protobuf.BoolValue by_banned = 3;
+
+        // Filters agents that can re-attest.
+        google.protobuf.BoolValue by_can_re_attest = 4;
```
Comment (faisal-memon) [inline proto/spire/api/server/agent/v1/agent.proto:82]: nit: Would eliminate the `_` between `re` and `attest` 

```suggestion
        google.protobuf.BoolValue by_can_reattest = 4;
```

Comment (guilhermocc) [reply proto/spire/api/server/agent/v1/agent.proto]: Thank you for the review, @faisal-memon! I've addressed this suggestion in 
990fa1d 😄

**Comment by faisal-memon** [comment]
lgtm

**Pull request merged**